plugins/jobs/src/js/data/JobModel.ts

```diff
@@ -271,8 +253,301 @@ type Query {
+    activeRuns(job: MetronomeClient.JobDetailResponse): JobRunConnection {
+      return typeResolvers.JobRunConnection(job.activeRuns);
+    },
+    docker(job: MetronomeClient.JobDetailResponse): Docker | null {
+      if (!job.run.docker) {
+        return null;
+      }
+
+      return {
+        forcePullImage: job.run.docker.forcePullImage,
+        image: job.run.docker.image
+      };
+    },
+    jobRuns(job: MetronomeClient.JobDetailResponse): JobRunConnection {
+      const { successfulFinishedRuns, failedFinishedRuns } = job.history;
+
+      const successfulFinishedRunsWithStatus: JobHistoryRunWithStatus[] = successfulFinishedRuns.map(
+        run => ({ ...run, status: "COMPLETED" as MetronomeClient.JobStatus }) // TODO: investiagte why we need to cast this
+      );
+
+      const failedFinishedRunsWithStatus: JobHistoryRunWithStatus[] = failedFinishedRuns.map(
+        run => ({
+          ...run,
+          status: "FAILED" as MetronomeClient.JobStatus
+        })
+      );
+
+      return typeResolvers.JobRunConnection([
+        ...job.activeRuns,
+        ...successfulFinishedRunsWithStatus,
+        ...failedFinishedRunsWithStatus
+      ]);
+    },
+    json(job: MetronomeClient.JobDetailResponse): string {
+      return JSON.stringify(cleanJobJSON(job));
+    },
+    labels(job: MetronomeClient.JobDetailResponse): Label[] {
+      return Object.entries(job.labels).map(([key, value]) => ({ key, value }));
+    },
+    lastRunStatus(job: MetronomeClient.JobDetailResponse): JobRunStatusSummary {
+      return typeResolvers.JobRunStatusSummary(job);
+    },
+    lastRunsSummary(job: MetronomeClient.JobDetailResponse): JobHistorySummary {
+      return typeResolvers.JobHistorySummary(job.history);
+    },
+    name(job: MetronomeClient.JobDetailResponse): string {
+      return job.id.split(".").pop() || "";
+    },
+    schedules(job: MetronomeClient.JobDetailResponse): ScheduleConnection {
+      return typeResolvers.ScheduleConnection(job.schedules);
+    },
+    scheduleStatus(job: MetronomeClient.JobDetailResponse): JobStatus {
+      const activeRuns = new JobRunList({ items: job.activeRuns });
+      const activeRunsLength = activeRuns.getItems().length;
+      const scheduleLength = job.schedules.length;
+
+      if (activeRunsLength > 0) {
+        const longestRunningActiveRun = activeRuns.getLongestRunningActiveRun();
+
+        return longestRunningActiveRun.status;
+      }
+
+      if (scheduleLength > 0) {
+        const schedule = job.schedules[0];
+
+        if (schedule != null && schedule.enabled) {
+          return "SCHEDULED";
+        }
+      }
+
+      if (scheduleLength === 0 && activeRunsLength === 0) {
+        return "UNSCHEDULED";
+      }
+
+      return "COMPLETED";
+    }
+  },
+  JobRun: {
+    jobID(
+      run: MetronomeClient.ActiveJobRun | MetronomeClient.JobHistoryRun
+    ): string {
+      if (isActiveRun(run)) {
+        return run.jobId;
+      }
+
+      if (isHistoryRun(run)) {
```

Comment:
Maybe here instead of the type guard just an else. If your run is guaranteed to be one of the two types, the else will infer right that it is a history Run, and accept the id. You will then not need the return "";

(I guess that works)

Reply:
Makes sense 👍